libdnf/base/transaction.cpp

```diff
@@ -372,6 +372,34 @@ std::string Transaction::all_package_solver_problems_to_string() {
     return output;
 }
 
+void Transaction::Impl::set_transaction_internal(
+    rpm::solv::GoalPrivate & solved_goal,
+    const solv::IdQueue & list_for_tsi,
+    TransactionPackage::Action action,
+    std::map<Id, std::vector<Id>> & replaced,
+    const std::string & no_obsoletes_msg) {
+    for (auto index = 0; index < list_for_tsi.size(); ++index) {
+        Id id = list_for_tsi[index];
+        auto obs = solved_goal.list_obsoleted_by_package(id);
+        if (obs.empty()) {
+            throw LogicError(no_obsoletes_msg);
```

Comment:
Are you sure this is a `LogicError`? I don't like it here. I understand if `obs` is empty it's an inconsistency in libsolv's output, but this is not something e.g. a user of the API can influence, it's just a libsolv bug. There is a huge amount of inconsistencies in libsolv's output which could occur, by which rule do we check some of them and not the others?

Do we really want to crash on this condition? I'd lean towards not checking this at all here, what do you think?

Reply:
Yes, I am sure that it is a logic Error. When it appears, it means something terribly wrong in libdnf or libsolv

Reply:
In dnf fore it ends in range exception in Python - 
`upgraded = obs.pop(0)`
`IndexError: pop from empty list`